blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticTokensIO.java

```diff
@@ -0,0 +1,291 @@
+package com.rackspacecloud.blueflood.io;
+
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Timer;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.ElasticClientManager;
+import com.rackspacecloud.blueflood.service.ElasticIOConfig;
+import com.rackspacecloud.blueflood.service.RemoteElasticSearchServer;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.Token;
+import com.rackspacecloud.blueflood.utils.GlobPattern;
+import com.rackspacecloud.blueflood.utils.Metrics;
+import org.apache.commons.lang.StringUtils;
+import org.elasticsearch.action.bulk.BulkRequestBuilder;
+import org.elasticsearch.action.index.IndexRequestBuilder;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.elasticsearch.index.query.QueryBuilders.*;
+
+/**
+ * {@link TokenDiscoveryIO} implementation using elastic search
+ */
+public class ElasticTokensIO implements TokenDiscoveryIO {
+
+    public static final String ES_DOCUMENT_TYPE = "tokens";
+
+    protected final Histogram batchHistogram = Metrics.histogram(getClass(), "Batch Sizes");
+    protected final Timer writeTimer = Metrics.timer(getClass(), "Write Duration");
+
+    protected final Timer esMetricNamesQueryTimer = Metrics.timer(getClass(), "ES Metric Names Query Duration");
```

Comment:
Can we keep the writeTimer and queryTimer names similar? I am fine with just "ES Write Duration" and "ES Query Duration".